src/Pact/Types/ChainMeta.hs

```diff
@@ -42,23 +43,33 @@ import Data.Word (Word64)
 
 -- internal pact modules
 
+import Pact.Parse
 import Pact.Types.ChainId (ChainId)
 import Pact.Types.Gas
 import Pact.Types.Util (AsString, lensyToJSON, lensyParseJSON)
 
 
 newtype EntityName = EntityName Text
-  deriving (Eq, Ord, NFData, ToJSON, FromJSON, Default, Generic, IsString, Serialize, Hashable, AsString)
+  deriving stock (Eq, Ord, Generic)
+  deriving newtype (Show, NFData, Hashable, Serialize, Default, ToJSON, FromJSON, IsString, AsString)
```

Comment:
deriving `Show` via `newtype` gets us the show instance removed below.

Reply:
Wow that might be the first concrete benefit beyond haskell snobbery I've seen for this stuff. Very cool